unstructured/partition/html/partition.py

```diff
@@ -31,6 +35,7 @@ def partition_html(
     ssl_verify: bool = True,
     skip_headers_and_footers: bool = False,
     detection_origin: Optional[str] = None,
+    contains_ontology_schema: bool = False,
```

Comment:
let's rename this parameter, and instead a boolean let's allow a string argument where None (or v1?) is the "classic" html parser. this will allow for additional "parsers" in the future without having to introduce a new parameter.

so to support the new parsing method, how about:

```
html_parser=v2
```